src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,156 @@
+        const fatherNameQuery = person.fatherName
+          ?.toLowerCase()
+          .includes(lowerCaseQuery);
+        const motherNameQuery = person.motherName
+          ?.toLowerCase()
+          .includes(lowerCaseQuery);
+
+        return nameQuery || fatherNameQuery || motherNameQuery;
+      });
+    }
+
+    if (century.length > 0) {
+      filteredPeople = filteredPeople.filter(person =>
+        century.includes(`${Math.ceil(person.born / 100)}`),
+      );
+    }
+
+    return filteredPeople;
+  };
+
+  const handleSort = () => {
+    const sort = searchParams.get('sort');
+    const order = searchParams.get('order');
+
+    const sortedPeople = handleFilter();
+
+    if (sort) {
+      switch (sort) {
+        case 'name':
+          if (!order) {
+            sortedPeople.sort((a, b) => a.name.localeCompare(b.name));
+
+            break;
+          }
+
+          sortedPeople.sort((a, b) => b.name.localeCompare(a.name));
+          break;
+
+        case 'sex':
+          if (!order) {
+            sortedPeople.sort((a, b) => a.sex.localeCompare(b.sex));
+
+            break;
+          }
+
+          sortedPeople.sort((a, b) => b.sex.localeCompare(a.sex));
+          break;
+
+        case 'born':
+          if (!order) {
+            sortedPeople.sort((a, b) => a.born - b.born);
+
+            break;
+          }
+
+          sortedPeople.sort((a, b) => b.born - a.born);
+          break;
+
+        case 'died':
+          if (!order) {
+            sortedPeople.sort((a, b) => a.died - b.died);
+
+            break;
+          }
+
+          sortedPeople.sort((a, b) => b.died - a.died);
+          break;
+
+        default:
+          return filterPeople;
+      }
+    }
+
+    return [...sortedPeople];
+  };
+
   return (
     <>
       <h1 className="title">People Page</h1>
 
       <div className="block">
         <div className="columns is-desktop is-flex-direction-row-reverse">
-          <div className="column is-7-tablet is-narrow-desktop">
-            <PeopleFilters />
-          </div>
+          {!loader && (
+            <div className="column is-7-tablet is-narrow-desktop">
+              <PeopleFilters handleFilter={handleFilter} />
+            </div>
+          )}
 
           <div className="column">
             <div className="box table-container">
-              <Loader />
+              {loader && <Loader />}
 
-              <p data-cy="peopleLoadingError">Something went wrong</p>
+              {false && (
+                <p data-cy="peopleLoadingError">Something went wrong</p>
+              )}
 
-              <p data-cy="noPeopleMessage">There are no people on the server</p>
+              {false && (
+                <p data-cy="noPeopleMessage">
+                  There are no people on the server
+                </p>
+              )}
 
-              <p>There are no people matching the current search criteria</p>
+              {false && (
```

Comment:
The same is here. But in this place you should check if some peoples exists in visible array after sorting.